src/components/App.js

```diff
@@ -1149,12 +1149,31 @@ class DashboardScreen extends React.Component {
 
     report.settings.league = this.props.leagues[4].id
 
+    this.generateDefaultReportName(report)
     this.setState({
       isCreatingReport: true,
       report
     })
   }
 
+  generateDefaultReportName (report) {
+    let nameMap = {};
+    this.props.reports.forEach(function (rep) {
+      nameMap[rep.settings.name] = true;
+    })
+
+    let defaultReportNumber = 0
+    let defaultReportName;
+    do {
+      defaultReportNumber++
+      defaultReportName = `${report.settings.league}-${defaultReportNumber}`
+    } while (nameMap[defaultReportName])
+    report.settings.name = defaultReportName
+    this.setState({
+      defaultReportName
```

Comment:
Why create a new state property? We could re-use the `report.settings.name` as the default value

Reply:
Ah! That is right, yes, there is a way but it is more hassle than just changing a value.

Essentially we'd have to initialize a report during the [`componentWillMount()`](https://reactjs.org/docs/react-component.html#componentwillmount) lifecycle function and then when the component mounts there _already exists_ a report object that it can update when the user works with the form.

Which would be a better approach.

However it isn't in the scope of this PR!